kafka/kafka.go

```diff
@@ -247,85 +244,68 @@ func (cm *consumerMessage) DiscardPayload() {
 	cm.cm = nil
 }
 
-func (ams *asyncMessageSource) ConsumeMessages(ctx context.Context, messages chan<- substrate.Message, acks <-chan substrate.Message) error {
+type consumerGroupHandler struct {
+	ctx context.Context
 
-	c, err := cluster.NewConsumerFromClient(ams.client, ams.consumerGroup, []string{ams.topic})
-	if err != nil {
-		return err
-	}
-
-	defer func() {
-		_ = c.Close()
-	}()
-
-	toAck := make(chan *consumerMessage)
-	eg, ctx := errgroup.WithContext(ctx)
-	eg.Go(func() error {
-		return ams.passMessagesToClient(ctx, c.Messages(), messages, toAck)
-	})
-	eg.Go(func() error {
-		return ams.passAcksToKafka(ctx, c, acks, toAck)
-	})
-
-	return eg.Wait()
+	messages chan<- substrate.Message
+	acks     <-chan substrate.Message
```

Comment:
Can we make this a 
```go
<-chan *consumerMessage
```
?

Reply:
No because as it is these channels are passed in by the user